models/models.js

```diff
@@ -14,4 +14,19 @@ exports.extractArticleById = (article_id) => {
         if (rows.length) return rows[0];
         else return Promise.reject({ errStatus: 404, msg: "Entry not found"});
     });
-};
\ No newline at end of file
+};
+
+exports.updateArticleVotesById = (article_id, body) => {
+    if ((!body.inc_votes) || (typeof body.inc_votes !== "number")) {
+        return Promise.reject({ errStatus: 400, msg: "Invalid request"});
+    } else {
+        return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
+        .then(({rows}) => {
+            if (rows.length) {
+                rows[0].votes += body.inc_votes;
+                return rows[0];
+             }
+            else return Promise.reject({ errStatus: 404, msg: "Entry not found"});
```

Comment:
CHALLENGE ACCEPTED

Reply:
So I had to delve deep into the murky world of SQL codes to make this work, but I presume this is what you were hinting at? Pretty impressed that it could be refactored that far, tbh